kernel/arch/dreamcast/kernel/entry.s

```diff
@@ -64,46 +65,31 @@ _irq_save_regs:
 	sts.l		pr,@-r0		! save PR   0x44
 	stc.l		spc,@-r0	! save PC   0x40
 	
-	add		#0x60,r0	! readjust register pointer
-	add		#0x44,r0
-	sts.l		fpul,@-r0	! save FPUL  0xe0
+	add		#0xA4,r0	! readjust register pointer
 	sts.l		fpscr,@-r0	! save FPSCR 0xdc
 	mov		#0,r2		! Set known FP flags
 	lds		r2,fpscr
-	fmov.s		fr15,@-r0	! save FR15  0xd8
-	fmov.s		fr14,@-r0	! save FR14
-	fmov.s		fr13,@-r0	! save FR13
-	fmov.s		fr12,@-r0	! save FR12
-	fmov.s		fr11,@-r0	! save FR11
-	fmov.s		fr10,@-r0	! save FR10
-	fmov.s		fr9,@-r0	! save FR9
-	fmov.s		fr8,@-r0	! save FR8
-	fmov.s		fr7,@-r0	! save FR7
-	fmov.s		fr6,@-r0	! save FR6
-	fmov.s		fr5,@-r0	! save FR5
-	fmov.s		fr4,@-r0	! save FR4
-	fmov.s		fr3,@-r0	! save FR3
-	fmov.s		fr2,@-r0	! save FR2
-	fmov.s		fr1,@-r0	! save FR1
-	fmov.s		fr0,@-r0	! save FR0   0x9c
-	frchg				! Second FP bank
-	fmov.s		fr15,@-r0	! save FR15  0x98
-	fmov.s		fr14,@-r0	! save FR14
-	fmov.s		fr13,@-r0	! save FR13
-	fmov.s		fr12,@-r0	! save FR12
-	fmov.s		fr11,@-r0	! save FR11
-	fmov.s		fr10,@-r0	! save FR10
-	fmov.s		fr9,@-r0	! save FR9
-	fmov.s		fr8,@-r0	! save FR8
-	fmov.s		fr7,@-r0	! save FR7
-	fmov.s		fr6,@-r0	! save FR6
-	fmov.s		fr5,@-r0	! save FR5
-	fmov.s		fr4,@-r0	! save FR4
-	fmov.s		fr3,@-r0	! save FR3
-	fmov.s		fr2,@-r0	! save FR2
-	fmov.s		fr1,@-r0	! save FR1
-	fmov.s		fr0,@-r0	! save FR0   0x5c
-	frchg				! First FP bank again
+	fschg                   ! Switch to pair FP moves
+    fmov.d  dr14,@-r0       ! save FR14 & FR15
+    fmov.d  dr12,@-r0       ! save FR12 & FR13
+    fmov.d  dr10,@-r0       ! save FR10 & FR11
+    fmov.d  dr8,@-r0        ! save FR8 & FR9
+    fmov.d  dr6,@-r0        ! save FR6 & FR7
+    fmov.d  dr4,@-r0        ! save FR4 & FR5
+    fmov.d  dr2,@-r0        ! save FR2 & FR3
+    fmov.d  dr0,@-r0        ! save FR0 & FR1
+    frchg                   ! Second FP bank
+    fmov.d  dr14,@-r0       ! save FR14 & FR15
+    fmov.d  dr12,@-r0       ! save FR12 & FR13
+    fmov.d  dr10,@-r0       ! save FR10 & FR11
+    fmov.d  dr8,@-r0        ! save FR8 & FR9
+    fmov.d  dr6,@-r0        ! save FR6 & FR7
+    fmov.d  dr4,@-r0        ! save FR4 & FR5
+    fmov.d  dr2,@-r0        ! save FR2 & FR3
+    fmov.d  dr0,@-r0        ! save FR0 & FR1
+    fschg                   ! Switch to single FP moves
+    frchg                   ! First FP bank again
```

Comment:
Formatting...

Reply:
Yes, I will be fixing the formatting after all the code changes (if any) at the end. Just so it will be easier to code review.